### DISCUSSION TITLE
Linking the testbed to the original source code

### DISCUSSION THREAD

**Discussion by rexxy-sasori** [cloudsimplus/cloudsimplus-testbeds#2] [Q&A] [answered]
Hi,

When I import the source code of cloudsimplus-testbed, the IDE shows that it uses the pre-compiled jar file of cloudsimplus posted on Mavens. Any suggestion on how I can make it depend on the cloudsimplus source code that I download on my own?
I want to keep this option open in case I need to modify the original source code. I tried it once by adjusting the project structure, but I got a compile error. 

How I attempted at achieving this:

1 - Removed  Maven: org.cloudsimplus:cloudsim-plus:7.3.0  originally present in the project structure of  cloudsimplus-testbeds and replaced it with the source code of cloudsimplus that I downloaded. 

2 - Restarted IDE and run one example in src/main/java/org/cloudsimplus/testbeds/sla/taskcompletiontime/CloudletTaskCompletionTimeWorkLoadMinimizationExperiment.java

3 - Saw an error on src/main/java/org/cloudsimplus/testbeds/sla/taskcompletiontime/CloudletTaskCompletionTimeWorkLoadMinimizationExperiment.java:121:24
java: incompatible types: int cannot be converted to org.cloudsimplus.builders.tables.TableColumn

<img width="1432" alt="Screenshot 2022-12-02 at 6 16 30 PM" src="https://user-images.githubusercontent.com/6620838/205270049-5ead550e-0cea-4c7e-84b6-67111fa7c832.png">

Any help is appreciated.

**Comment by manoelcampos** [comment, marked as answer]
You just need to execute `mvn clean install` inside the root directory or your cloudsimplus clone.
If you have downloaded cloudsimplus and cloudsimplus-testbeds, on IntelliJ, you can open the directory containing both projects that IntelliJ enables you to build both projects and use the built version.

**Reply by manoelcampos** [reply]
Anyway, if you are changing the code inside cloudsimplus repository, you probably are doing something wrong. 
This is one of the many mistakes CloudSim users do.
CloudSim Plus is a reengineered project which is really extensible. 
If you want to implement something, you just need to create a new project, add cloudsimplus as a dependency and create your classes that extend cloudsimplus ones.

**Reply by rexxy-sasori** [reply]
Thanks for the quick response. I may forgot to mention that somehow the example repo can be recognized by IntelliJ successfully. Anyway, I do agree a simple extension may be a much easier alternative after failingly trying different things.
